components/SurveyModal.js

```diff
@@ -0,0 +1,566 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+// Callback function for user to complete mailchimp signup
+function newsletterCallback() {
+  const mailchimpForm = document.getElementById('mailchimp-form');
```

Comment:
Is fetching an element by ID like this a smell? It feels like the "React way" to tie behavior to an element would be to use a component. Or if that doesn't work for some reason, a [`ref`](https://reactjs.org/docs/refs-and-the-dom.html).

Reply:
Agreed, I've updated this to use ref. I think we could probably rework our basic MailChimp component some to be more flexible, but I think I will save that for another time.

Reply:
Awesome! 👏